assets/js/src/settings/store/actions.ts

```diff
@@ -25,3 +30,138 @@ export function* saveSettings() {
   yield { type: 'TRACK_SETTINGS_SAVED' };
   return { type: 'SAVE_DONE' };
 }
+
+export function* verifyMssKey(key: string, activateMssIfKeyValid: boolean) {
+  const { success, error, res } = yield {
+    type: 'CALL_API',
+    endpoint: 'services',
+    action: 'checkMSSKey',
+    data: { key },
+  };
+  if (!success) {
+    return updateKeyActivationState({
+      mssStatus: 'invalid',
+      mssMessage: error.join(' ') || null,
+    });
+  }
+  const fields: Partial<KeyActivationState> = {
+    mssMessage: res.data.message || null,
+  };
+  if (activateMssIfKeyValid) {
+    const call = yield {
+      type: 'CALL_API',
+      endpoint: 'settings',
+      action: 'set',
+      data: {
+        mta_group: 'mailpoet',
+        mta: { method: 'MailPoet', mailpoet_api_key: key },
+        signup_confirmation: { enabled: '1' },
+      },
+    };
```

Comment:
Are we doing a bit of duplicated work here? Shouldn't this be managed only by "setSetting" and save?

Reply:
Good point. Unfortunately, I can't do that because the `saveSettings` action will do additional things that we don't want here (track mixpanel event), and also we will not know if it succeeded or failed (an action creator should not access the state directly unless there is no other choice).